### PR TITLE
Location page x Learn tooltip

### PR DESCRIPTION
**Pull request by chasulin**
**WIP, do not merge** (Replaces #2703)

[Doc with tooltip copy
](https://docs.google.com/document/d/1rukcHn6yPNPE16iYX5AGeEXDr3fSwpqtwN6OVxIRDFA/edit)[Doc with links for 'data sources' tooltip
](https://docs.google.com/spreadsheets/d/1b3zMFadzPlq5-h-JHuwE2UC-MhhyQHAv6f74wL0JwMY/edit#gid=0)

Still todo:
- [x] Add tracking to tooltips

### CODE REVIEW

File: src/components/Explore/Explore.style.ts [hunk 1]
```diff
@@ -356,3 +356,22 @@ export const NormalizeSubLabel = styled.div`
   margin-left: 2rem;
   transform: translateY(-0.75rem);
 `;
+
+export const DisclaimerWrapper = styled.div`
```
Comment (chasulin) [inline src/components/Explore/Explore.style.ts:360]: these were originally in `components/Disclaimer/Disclaimer.style` and used by explore. theyre no longer used by the disclaimer so i moved them here

File: public/admin/config.yml [hunk 2]
```diff
@@ -798,7 +798,7 @@ collections:
     files:
       - name: Info tooltips
         label: Info tooltips
-        file: src/cms-content/info-tooltips.json
+        file: src/cms-content/tooltips/info-tooltips.json
```
Comment (chasulin) [inline public/admin/config.yml:801]: re-did the original setup, in which i forgot to name a folder and so the files went directly into `cms-content`-- hence the deleted files below (`src/cms-content/info-tooltips.json`, `src/cms-content/metric-calculation-tooltips.json`)

i also renamed the outer folder from `cms-content/infoTooltips` --> `cms-content/tooltips`, hence the deleted `src/cms-content/infoTooltips/index.ts`

😬 😬

**Review by pnavarrc — COMMENTED** [review]
Some comments from Thursday

**Review by pnavarrc — APPROVED** [review]
Looks good! I added some notes on how to structure functions and components, but we can discuss those suggestions and maybe implement them on a follow up PR. Some notes on accessibility:

- The disclaimer shows when the text with the tooltip is focused, but since the content has a link, the user will need to hit <kbd>tab</kbd> twice to get to the second span

recording of the tooltip behaviour using a keyboard to focus

- Since the tooltip opens on focus, when the user tries to activate the button nothing happens. VoiceOver reads the content of the tooltip and if the user hits tab, they reach the link in the tooltip before moving on to the next focusable element.

We can try to make the experience a bit smoother after launch, but as it is, a screen reader user can open the content, visit the link and access the tooltip content. A sighted user using the keyboard only would have trouble going to the link inside the tooltip, because when the link is focused, the tooltip is closed.

File: src/components/InfoTooltip/index.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+export type StyledTooltipProps = Omit<TooltipProps, 'children'> & {
+  mainCopy?: string;
+};
+
+export function renderTooltipContent(body: string): React.ReactElement {
+  return (
+    <>
+      <StyledMarkdown source={body} />
+    </>
```
[the remark this answers is not in the document]
Comment (chasulin) [reply src/components/InfoTooltip/index.tsx:17]: ah yes, thats a remnant. removing!

File: src/components/InfoTooltip/DescriptionDiv.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  content: React.ReactNode;
+}> = ({ content }) => {
+  return (
+    <HiddenDiv role="tooltip" id="tooltip-description">
+      {content}
+    </HiddenDiv>
+  );
+};
+
+export default DescriptionDiv;
```
[the remark this answers is not in the document]
Comment (chasulin) [reply src/components/InfoTooltip/DescriptionDiv.tsx:21]: punting to a later pr

File: src/components/LocationPage/ChartBlock.tsx [hunk 5]
```diff
@@ -64,7 +68,7 @@ function ChartBlock(props: {
       {hasMetric && (
         <>
           <MetricChart metric={metric} projections={projections} />
-          <Disclaimer metricName={metric} projections={projections} />
+          <Disclaimer metricName={metric} region={region} />
```
[the remark this answers is not in the document]
Comment (chasulin) [reply src/components/LocationPage/ChartBlock.tsx:71]: hm [theres a key in ChartsHolder](https://github.com/covid-projections/covid-projections/blob/808d3bc4d3444b322c10d15e26197604b86c6330/src/components/LocationPage/ChartsHolder.tsx#L178) on the whole component. i dont think we need one here?

File: src/cms-content/tooltips/index.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+function getTooltipContentById(id: string, tooltipSet: Tooltip[]): Tooltip {
+  const content = tooltipSet.find((item: Tooltip) => item.id === id);
+  assert(content, `Info tooltip content unexpectedly not found for id: ${id}`);
+  return content;
+}
+
+interface MetricTooltipObj {
+  metricDefinition: Tooltip;
+  metricCalculation: Tooltip;
+}
```
[the remark this answers is not in the document]
Comment (chasulin) [reply src/cms-content/tooltips/index.ts:23]: punting

File: src/components/Disclaimer/Disclaimer.tsx [hunk 7]
```diff
@@ -1,20 +1,18 @@
...
   metricName: number;
-  projections: Projections;
+  region: Region;
 }) => {
   return (
     <DisclaimerWrapper>
-      <DisclaimerBody>
-        {getMetricDisclaimer(metricName, projections)}
-      </DisclaimerBody>
+      {getMetricDisclaimer(metricName, region)}
```
[the remark this answers is not in the document]
Comment (chasulin) [reply src/components/Disclaimer/Disclaimer.tsx:15]: also punting

**Pull request merged**